lib/embulk/filter/ruby_proc.rb

```diff
@@ -175,17 +192,25 @@ def add(page)
           end
 
           record_hashes.each do |record_hash|
-            procs.each do |col, pr|
-              next unless record_hash.has_key?(col)
-              next if record_hash[col].nil? && skip_nils[col]
+            begin
+              skip_row_procs.each do |pr|
+                raise SkipRecordException if pr.call(record_hash)
```

Comment:
I think that this feature needs not to raise Exception.
It is non-local exit by condition.
To use throw-catch is better than to use Exception.

Especially in JRuby, Exception is high cost.
If non-local exit often occurs,  throw-catch is slightly fast.

Reply:
Right. I rewrote the patch. I worried that thow-catch is rare feature in Ruby.